### PR TITLE
Explicitly list what methods are included in unmounting and remounting

### PR DESCRIPTION
**Pull request by eps1lon**
[Preview](https://reactjs-60pzl97ut-fbopensource.vercel.app/docs/strict-mode.html#ensuring-reusable-state)
To help discoverability of what Strict Effects affects (e.g. CTRL+f "componentDidMount").
Originally reported in  https://github.com/facebook/react/issues/24467#issuecomment-1114185870

### CODE REVIEW

File: content/docs/strict-mode.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ In the above example, strict mode checks will *not* be run against the `Header`
 * [Warning about deprecated findDOMNode usage](#warning-about-deprecated-finddomnode-usage)
 * [Detecting unexpected side effects](#detecting-unexpected-side-effects)
 * [Detecting legacy context API](#detecting-legacy-context-api)
-* [Detecting unsafe effects](#detecting-unsafe-effects)
+* [Ensuring reusable state](#ensuring-reusable-state)
```
Comment (eps1lon) [inline content/docs/strict-mode.md:24]: Drive-by fix. Currently the fragment link is broken: https://reactjs.org/docs/strict-mode.html#detecting-unsafe-effects

**Review by gaearon — APPROVED** [review]
